components/tikv_alloc/src/lib.rs

```diff
@@ -128,27 +128,20 @@ mod imp {
         String::from_utf8_lossy(&buf).into_owned()
     }
 
-    pub struct JemallocStats {
-        pub allocated: usize,
-        pub active: usize,
-        pub metadata: usize,
-        pub resident: usize,
-        pub mapped: usize,
-        pub retained: usize,
-    }
+    pub type AllocStats = Vec<(&'static str, usize)>;
```

Comment:
Perhaps place this outside of `mod imp`?

Reply:
This assumes that all the metric values are Gauge type, I worry this may be wrong later if we introduce other allocators.

Reply:
Then `AllocStatsCollector` will also need to change to not use an `IntGaugeVec`.

Reply:
Yes. Once there are more stat types the abstraction here will need to change. Offhand I'd expect the i64's to change to a custom enum for each possible type.